lib/plans/store_checkout.dart

```diff
@@ -0,0 +1,192 @@
+                ),
+                const SizedBox(height: 20.0),
+                CText("email_hint_pro".i18n, style: tsBody1),
+                const SizedBox(height: 24.0),
+                SizedBox(
+                  width: double.infinity,
+                  child: Button(
+                    text: "continue".i18n,
+                    onPressed: () {
+                      state = _CheckOutState.withEmail;
+                      _validateEmailAndContinue();
+                    },
+                  ),
+                ),
+                const SizedBox(height: 16.0),
+                Center(
+                  child: TextButton(
+                      onPressed: () {
+                        state = _CheckOutState.withoutEmail;
+                        startPurchaseFlow();
+                      },
+                      child: CText(
+                        "continue_without_email".i18n.toUpperCase(),
+                        style: tsButtonPink,
+                      )),
+                )
+              ],
+            ),
+          );
+        }));
+  }
+
+  void _validateEmailAndContinue() {
+    if (emailFieldKey.currentState?.validate() == false) {
+      return;
+    }
+    if (emailController.text.isEmpty) {
+      showError(context, error: 'please_enter_a_valid_email_address'.i18n);
+      return;
+    }
+
+    startPurchaseFlow();
+  }
+
+  void startPurchaseFlow() {
+    FocusManager.instance.primaryFocus?.unfocus();
+    if (Platform.isAndroid) {
+      submitPlayPayment();
+      return;
+    }
+    _proceedToCheckoutIOS();
+  }
+
+  void submitPlayPayment() async {
+    try {
+      context.loaderOverlay.show();
+      // Await the result of the payment submission.
+      await sessionModel.submitPlayPayment(
+          widget.plan.id, emailController.value.text);
+      context.loaderOverlay.hide();
+      resolveRoute();
+    } catch (error, stackTrace) {
+      // In case of an error, hide the loader and show the error message.
+      context.loaderOverlay.hide();
+      showError(context, error: error, stackTrace: stackTrace);
+    }
+  }
+
+  void _proceedToCheckoutIOS() {
+    final appPurchase = sl<AppPurchase>();
+    final email = (state == _CheckOutState.withEmail ? emailController.text : "");
+    // Just as safe check
+    if (email.isNotEmpty && !EmailValidator.validate(email)) {
+      showError(context, error: 'please_enter_a_valid_email_address'.i18n);
+      return;
+    }
+    try {
+      context.loaderOverlay.show();
+      appPurchase.startPurchase(
+        email: email,
+        planId: widget.plan.id,
+        onSuccess: () {
+          context.loaderOverlay.hide();
+          resolveRoute();
+        },
+        onFailure: (error) {
+          context.loaderOverlay.hide();
+          CDialog.showError(
+            context,
+            error: error,
+            description: error.toString(),
+          );
+        },
+      );
+    } catch (error) {
+      context.loaderOverlay.hide();
```

Comment:
Are we able to put `context.loaderOverlay.hide();` in a finally to avoid having to repeat it several times or no?